extensions/wikia/AdEngine/js/provider/recirculation.js

```diff
@@ -0,0 +1,35 @@
+/*global define*/
+define('ext.wikia.adEngine.provider.recirculation', [
+	'wikia.document',
+	'wikia.log',
+	'ext.wikia.adEngine.adContext',
+	'ext.wikia.adEngine.taboolaHelper',
+	require.optional('ext.wikia.recirculation.recirculation')
+], function (doc, log, adContext, taboolaHelper, recirculation) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.provider.recirculation',
+		recirculationSlot = 'RECIRCULATION_RAIL';
+
+	function canHandleSlot(slotName) {
+		var properSlot = slotName === recirculationSlot;
+		log(['canHandleSlot', slotName, (properSlot && recirculation)], 'debug', logGroup);
+
+		return (properSlot && recirculation);
+	}
+
+	function fillInSlot(slotName, slotElement, success) {
+		log(['fillInSlot', slotName, slotElement], 'debug', logGroup);
+
+		taboolaHelper.loadTaboola();
```

Comment:
I don't think you need to load Taboola here because it's done as a part of injectRecirculationModule. (Also this means there's no need to expose loadTaboola method).